packages/@aws-cdk/core/lib/token.ts

```diff
@@ -150,6 +151,11 @@ export interface ResolveOptions {
    * The resolver to apply to any resolvable tokens found
    */
   readonly resolver: ITokenResolver;
+
+  /**
+   * Whether the resolution is being executed during the prepare phase or not.
```

Comment:
Add a comment explaining what is the implication of setting this to `true` and also add a "@default" value.